src/mc-fle2-find-range-payload.c

```diff
@@ -112,6 +112,7 @@ mc_FLE2FindRangePayload_serialize (const mc_FLE2FindRangePayload_t *payload,
          if (!bson_append_document_end (&g_bson, &etc_bson)) {
             return false;
          }
+         if (g_index == UINT32_MAX) break;
```

Comment:
Appending 2 billion documents would actually be impossible, since it will hit the BSON size limit first, so something would have gone very wrong for this to end up at UINT32_MAX